### PR TITLE
CSHARP-4267: Updated to latest transaction spec tests and enabled load balanced transaction spec tests.

### PR DESCRIPTION
**Pull request by JamesKovacs**

### CODE REVIEW

File: tests/MongoDB.Driver.Tests/Specifications/transactions/TransactionUnifiedTestRunner.cs [hunk 1]
```diff
@@ -38,11 +38,6 @@ public TransactionUnifiedTestRunner(ITestOutputHelper testOutputHelper)
         [ClassData(typeof(TestCaseFactory))]
         public void Run(JsonDrivenTestCase testCase)
         {
-            if (testCase.Name.Contains("mongos-unpin.json"))
-            {
-                throw new SkipException("Load balancer support not yet implemented.");
-            }
-
             using (var runner = new UnifiedTestRunner(loggerFactory: LoggerFactory))
```
Comment (JamesKovacs) [inline tests/MongoDB.Driver.Tests/Specifications/transactions/TransactionUnifiedTestRunner.cs:41]: We were still skipping the load balanced transaction tests even though load balancer support has been implemented.

**Review by DmitryLukyanov — APPROVED** [review]
LGTM, but please launch LB tests too in the EG patch in this PR

**Comment by JamesKovacs** [comment]
> LGTM, but please launch LB tests too in the EG patch in this PR
Good point. I added our LB variants to the patch build. I'll wait for them to go green before merging.

**Comment by JamesKovacs** [comment]
Load balanced variants are all green. Proceeding with the merge.

**Pull request merged**